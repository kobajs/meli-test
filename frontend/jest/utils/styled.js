import { shallow, mount } from "enzyme";
import ThemeProvider from "../../src/theme/ThemeProvider";

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(withTheme(children), options);

  return wrapper;
}

export function withTheme(children) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments);
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments);
}
