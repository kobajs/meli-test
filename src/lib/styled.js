import get from 'lodash/get'

export function theme(path) {
  return props => get(props, `theme.${path}`);
}

export function breakpoints(breakpoint) {
  return props => `@media (min-width: ${props.theme.breakpoints[breakpoint]}px)`;
}