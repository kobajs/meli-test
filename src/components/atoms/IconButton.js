import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import get from "lodash/get";

const Button = styled.button`
  border: 0px;
  height: 40px;
  width: 40px;
  background-color: ${({ theme, backgroundColor }) =>
    get(theme.palette, backgroundColor)};
`;

function IconButton({ type, backgroundColor, Icon }) {
  return (
    <Button type={type} backgroundColor={backgroundColor}>
      <img src={Icon} />
    </Button>
  );
}

IconButton.propTypes = {
  Icon: PropTypes.node.isRequired,
  type: PropTypes.string,
  backgroundColor: PropTypes.string
};

IconButton.defaultProps = {
  type: "button",
  backgroundColor: "transparent"
};

export default IconButton;
