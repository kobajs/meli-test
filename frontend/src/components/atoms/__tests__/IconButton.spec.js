import React from "react";
import renderer from "react-test-renderer";

import IconButton from "../IconButton";
import { withTheme } from "../../../../jest/utils/styled";

it("renders correctly", () => {
  const ThemedIconButton = withTheme(
    <IconButton Icon="/static/logo.png">Test IconButton</IconButton>
  );
  const tree = renderer.create(ThemedIconButton).toJSON();
  expect(tree).toMatchSnapshot();
});
