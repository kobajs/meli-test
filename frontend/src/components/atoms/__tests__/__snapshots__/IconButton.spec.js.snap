// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="IconButton__Button-yvsbn1-0 hwvofK"
  type="button"
>
  <img
    src="/static/logo.png"
  />
</button>
`;
